### ISSUE TITLE
Ensure instrumentation field errors are populated in FetchedResult

### ISSUE THREAD

**Issue by schenkman** [graphql-java/graphql-java#2519]
Hi team! I am part of the Core API team at Twitter and we are currently in a large project migrating from Sangria to graphql-java to power Twitter's graphql infrastructure. We've been meaning to introduce ourselves and figured a PR would be the best way to do so.

Currently if a DataFetcher raises an exception in any way, the FetchedValue given in the instrumentation does not contain any errors information. The only place errors are available are on the ExecutionContext within a list, which is not ideal to determine if the current field has an error as we would need to iterate the list every time and that wouldn't be performant.

The attached PR has more information.

**Comment by schenkman** [comment]
#2520

**Comment by bbakerman** [comment]
I think this went out in 17.2 btw

**Comment by schenkman** [comment]
Yeah totally, we already pulled it in and started using it, which is great. Thank you!
